models/classes/runner/time/QtiTimer.php

```diff
@@ -441,33 +444,41 @@ public function load()
         if (isset($data)) {
             if (!is_array($data)) {
                 $data = [
-                    self::STORAGE_KEY_TIME_LINE => $data,
+                    QtiTimeStorageFormat::STORAGE_KEY_TIME_LINE => $data,
                 ];
             }
 
-            if (isset($data[self::STORAGE_KEY_TIME_LINE])) {
-                $this->timeLine = $this->unserializeTimeLine($data[self::STORAGE_KEY_TIME_LINE]);
+            if (isset($data[QtiTimeStorageFormat::STORAGE_KEY_TIME_LINE])) {
+                $this->timeLine = $this->unserializeTimeLine($data[QtiTimeStorageFormat::STORAGE_KEY_TIME_LINE]);
             } else {
                 $this->timeLine = new QtiTimeLine();
             }
 
-            if (isset($data[self::STORAGE_KEY_EXTRA_TIME])) {
-                $this->extraTime = $data[self::STORAGE_KEY_EXTRA_TIME];
+            if (isset($data[QtiTimeStorageFormat::STORAGE_KEY_EXTRA_TIME])) {
+                $this->extraTime = $data[QtiTimeStorageFormat::STORAGE_KEY_EXTRA_TIME];
             } else {
                 $this->extraTime = 0;
             }
 
-            if (isset($data[self::STORAGE_KEY_EXTENDED_TIME])) {
-                $this->extendedTime = $data[self::STORAGE_KEY_EXTENDED_TIME];
+            if (isset($data[QtiTimeStorageFormat::STORAGE_KEY_EXTENDED_TIME])) {
+                $this->extendedTime = $data[QtiTimeStorageFormat::STORAGE_KEY_EXTENDED_TIME];
             } else {
                 $this->extendedTime = 0;
             }
-            if (isset($data[self::STORAGE_KEY_CONSUMED_EXTRA_TIME])) {
-                $this->consumedExtraTime = $data[self::STORAGE_KEY_CONSUMED_EXTRA_TIME];
+            if (isset($data[QtiTimeStorageFormat::STORAGE_KEY_CONSUMED_EXTRA_TIME])) {
+                $this->consumedExtraTime = $data[QtiTimeStorageFormat::STORAGE_KEY_CONSUMED_EXTRA_TIME];
             } else {
                 $this->consumedExtraTime = 0;
             }
 
+            if (isset($data[QtiTimeStorageFormat::STORAGE_KEY_TIMER_ADJUSTMENT_MAP])) {
+                $this->adjustmentMap = $this->unserializeAdjustmentMap(
+                    $data[QtiTimeStorageFormat::STORAGE_KEY_TIMER_ADJUSTMENT_MAP]
+                );
+            } else {
```

Comment:
If we already assign `$this->adjustmentMap = new AdjustmentMap();` in the constructor I don't think we still need this `else` block.

Reply:
The storage can be changed during runtime with `QtiTimer::setStorage(...)` and if this new storage will not have a stored adjustment map then there might be stale data left in `$this->adjustmentMap`.